akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -162,23 +162,20 @@ private[akka] class AffinityPool(
         indexFor(runnableHash)
       else {
         @tailrec
-        def updateAndOrGetIndex(): Int = {
-          val prev = hashCache.get()
-          if (prev.size > fairDistributionThreshold) indexFor(runnableHash)
```

Comment:
@zaharidichev You'll definitely want to rerun the AffinityPoolBench with this latest commit to the wip-affinity-√ branch!